src/net_processing.cpp

```diff
@@ -3447,6 +3433,16 @@ bool ProcessMessage(CNode& pfrom, const std::string& msg_type, CDataStream& vRec
     }
 
     if (msg_type == NetMsgType::FILTERLOAD) {
+        if (!(pfrom.GetLocalServices() & NODE_BLOOM)) {
```

Comment:
This code block is repeated 3 times and bloats the `ProcessMessage` function. Would a helper function like `MaybePunishNodeForBloom` be appropriate or is this fine?

Reply:
Yes, but I think you can also just reduce the complexity here. I don't think there's any benefit to calling `Misbehaving()` on a peer that sends us a `FILTER*` message. Calling `Misbehaving(100)` results in us disconnecting the peer and then marking it discouraged (see #19219). Discouragement is a mechanism to prevent our limited connection slots being used up by incompatible or broken peers.

So I think we should just set `fDisconnect` for any peer that sends us a `FILTER*` message when we don't have `NODE_BLOOM` enabled, regardless of their version number.

Reply:
In theory it is possible to set `-banscore=101` to not disconnect a 100-misbehaving peer. Though, that would only kick down the can by one more message. I can't see a use case for that and encouraging users to change banscore to accomodate misbehaving light clients seems ill-advised. So I agree with the suggestion made by @jnewbery .

Reply:
We should remove `-banscore` entirely. It's not useful user configuration.

Reply:
Thanks for the suggestion! I've updated it but I think Travis timed out on one of the builds.